### PR TITLE
Update RESTinio to version 0.7.0

### PR DESCRIPTION
**Pull request by ngrodzitski**
Specify library name and version:  **RESTinio/0.7.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — APPROVED** [review]
Usually if the recipes do not diverge too much we usually prefer to keep them as one, this would be in the limit - thanks a lot for taking the time to fix it for the new version, really appreciated! :)

File: recipes/restinio/all/conanfile.py [hunk 1]
```diff
@@ -64,51 +59,44 @@ def package_id(self):
...
+
+    def build(self):
+        pass
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def package(self):
+        copy(self, "*.*pp", src=os.path.join(self.source_folder, "dev", "restinio"), dst=os.path.join(self.package_folder, "include", "restinio"))
         copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```
Comment (RubenRBS) [inline recipes/restinio/all/conanfile.py:94]: Is CMAKE.install() not suitable anymore?

File: recipes/restinio/all/conanfile.py [hunk 2]
```diff
@@ -64,51 +59,44 @@ def package_id(self):
...
+            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+
+    def build(self):
+        pass
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def package(self):
+        copy(self, "*.*pp", src=os.path.join(self.source_folder, "dev", "restinio"), dst=os.path.join(self.package_folder, "include", "restinio"))
```
Comment (RubenRBS) [inline recipes/restinio/all/conanfile.py:93]: is this packaging both cpp and hpp files as a header-library?

**Review by RubenRBS — COMMENTED** [review]
Thanks! I have a question regarding the new package() method, but otherwise looks great!

File: recipes/restinio/all/conanfile.py (hunk 1 above)
Comment (ngrodzitski) [reply recipes/restinio/all/conanfile.py:94]: I could have use it, yes, but it goes with maintenance costs:
* set the tuning variables right (they can be added/removed and their names/values might be changed in future versions) 
* Conan itself may change the cmake-routines (like v1=>v2).

On the other hand RESTinio takes a promise that a set of headers would be enough - so copying sources is a nice way to go without extra cmake machinery.

File: recipes/restinio/all/conanfile.py (hunk 2 above)
Comment (ngrodzitski) [reply recipes/restinio/all/conanfile.py:93]: > is this packaging both cpp and hpp files as a header-library?

"*.*pp" is to also include "*.ipp" files.

**Comment by uilianries** [comment]
Thank you for checking this breaking change!

It's a bit hard to review it, because is actually changing the previous, and copying the old version too. My suggestion it only adding a new folder (e.g 0.7) and include it in config.yml, so it will generate less noise and a clean review.

**Review by jwillikers — APPROVED** [review]

**Comment by fdgStilla** [comment]
We would be interested in using it with Visual2019 but you limit it to Visual2022 (VS17). VS17 is actually equivalent to msvc193 so it weird to limit to VS17 and msvc191. What is the actual limit for this version?
```
            "Visual Studio": "17",
            "msvc": "191"
```

**Comment by eao197** [comment]
I haven't checked v.0.7.0, but I've checked v.0.7.1 right now and it compiles and passes tests under VS2019.

**Pull request merged**